cpp/velox/memory/VeloxMemoryManager.h

```diff
@@ -100,20 +103,24 @@ class VeloxMemoryManager final : public MemoryManager {
  private:
   bool tryDestructSafe();
 
+  void dropMemoryPool(const std::string& name);
+
 #ifdef GLUTEN_ENABLE_HBM
   std::unique_ptr<VeloxMemoryAllocator> wrappedAlloc_;
 #endif
 
-  // This is a listenable allocator used for arrow.
-  std::unique_ptr<MemoryAllocator> listenableAlloc_;
   std::unique_ptr<AllocationListener> listener_;
   std::unique_ptr<AllocationListener> blockListener_;
-  std::unique_ptr<arrow::MemoryPool> arrowPool_;
+
+  std::shared_ptr<ArrowMemoryPool> defaultArrowPool_;
+  std::unordered_map<std::string, std::weak_ptr<ArrowMemoryPool>> arrowPools_;
 
```

Comment:
It's understandable to use weak pointers, but I am curious how many instance will be added to this map in the worst case?

Reply:
Not sure. But for long running spark process, a new pool will be created for every shuffle writer, which might be a large number. If we don't use weak pointer here, we still need to delete the pool if it won't be used.